ocaml/xapi-cli-server/cli_operations.ml

```diff
@@ -7681,19 +7641,64 @@ let update_resync_host _printer rpc session_id params =
   let host = Client.Host.get_by_uuid ~rpc ~session_id ~uuid in
   Client.Pool_update.resync_host ~rpc ~session_id ~host
 
-let host_apply_updates _printer rpc session_id params =
+let print_available_updates ~rpc ~session_id ~fd ~printer ~host =
+  let task =
+    Client.Task.create ~rpc ~session_id
+      ~label:
+        (Printf.sprintf "Get updates for host (ref=%s)" (Ref.string_of host))
+      ~description:""
+  in
+  Client.Task.set_progress ~rpc ~session_id ~self:task ~value:(-1.0) ;
+  let uri =
+    Printf.sprintf "%s?session_id=%s&task_id=%s&host_refs=%s"
+      Constants.get_updates_uri (Ref.string_of session_id) (Ref.string_of task)
+      (Ref.string_of host)
+  in
+  finally
+    (fun () ->
+      printer (Cli_printer.PMsg "Get available updates:") ;
+      marshal fd (Command (PrintHttpGetJson uri)) ;
+      let response = ref (Response Wait) in
+      while !response = Response Wait do
+        response := unmarshal fd
+      done ;
+      let ok =
+        match !response with
+        | Response OK ->
+            true
+        | Response Failed ->
+            if Client.Task.get_progress ~rpc ~session_id ~self:task < 0.0 then
```

Comment:
Again: `<=` is probably safer

Reply:
Similarly, this is for the error category that the CLI client has not contacted with the server. Added a comment for this.